### PR TITLE
[ready] refactor: enumerate

### PR DESCRIPTION
**Pull request by roelofvandijk**
Use enumerate or direct iteration instead of `range(len(abc)`

### CODE REVIEW

**Comment by tinyb0t** [comment]
Changes made in `tinygrad/`:
```
------------------------------------------------------------
files                             insertions       deletions
------------------------------------------------------------
tinygrad/codegen/kernel.py                 2               2
tinygrad/codegen/optimizer.py              2               2
tinygrad/lazy.py                           1               1
tinygrad/nn/image.py                       1               1
tinygrad/runtime/ops_cpu.py                1               1
tinygrad/tensor.py                         4               4
------------------------------------------------------------
total                                     11              11
------------------------------------------------------------
lines added in the tinygrad folder: 0
```

File: tinygrad/tensor.py [hunk 1]
```diff
@@ -316,8 +316,8 @@ def num_zeros(step, dim_sz): return 0 if step == 1 or (y := dim_sz % step) == 0
         dim_shape = next(it_shape)
         if isinstance(s, slice): final_shape.append(dim_shape)
         elif tensor_found:
-          for i_ in range(len(tensor_found)):
-            if tensor_found[i_][0] > i: sub[i_] -= 1
```
Comment (wozeparrot) [inline tinygrad/tensor.py:320]: Could probably revert the change here, because `__getitem__` is refactored in #1661

Comment (roelofvandijk) [reply tinygrad/tensor.py]: @wozeparrot Done.
